word_counter.rb

```diff
@@ -0,0 +1,18 @@
+class WordCounter
+  def self.count file  
+    raise 'WordCounter: no file given' unless file 
+    # use inject to create a hash of word counts
+    to_word_array(file).inject({}) { |h,w| h[w] ? h[w] += 1 : h[w] = 1; h }
```

Comment:
I purposefully chose to chain method calls to avoid the cost of unnecessary variable assignment. I do agree that more verbose loop variable naming makes for more readable code though.